### DISCUSSION TITLE
Totally different variable value when using FXs

### DISCUSSION THREAD

**Discussion by WayneX-Wong** [idaholab/moose#27782] [Q&A Modules: General]
Hello,

I'm learning to use FXs, and here's the model, a cylinder with 8 slots on the edge, with each surface of the slots applied a Dirichlet BCs, all the other surfaces applied 0-flux Neumann BCs.

I Use `FXVolumeUserObject` to get a functional expansion of temperature distribution (Use AuxVariable `T_FXS` to get UserObject result), input file be like
[main.txt](https://github.com/user-attachments/files/15541985/main.txt)

The calculation converge well, but the expansional T_FXs differs greatly with solved T, shown as below:
<img width="552" alt="pic1" src="https://github.com/idaholab/moose/assets/111510550/be82249c-200d-4b70-8a76-7e5169abf4c4">

What could be the reason for this?

**Comment by GiudGiud** [comment]
Hello

First you should try with a cylindrical mesh. Once it works there you can move on to this more interesting shape. 

Functoonal expansion suffer a lot from being used on a geometry that does not match what they were derived for. I have seen some large deviations from simply using a first order coarse cylinder 

Guillaume

**Reply by WayneX-Wong** [reply]
Hello Guillaume,

I try with a simple cylinder just now, the functional expansion result is well consistent with calculated one. I didn't change any FXs-related input.

While I'm still not sure what that says with my problem. Do you have any further suggestions?

Thanks for your reply!

**Reply by GiudGiud** [reply]
My guess at this point is that when you use a non cylindrical shape to perform the integral, you get the completely wrong answer because you are effectively integrating a 0 value in the parts of the cylinder that do not overlap with the geometry. 

Another side effect that may be relevant is that the polynomials are not orthogonal on this new domain. So the projection is not as straightforward anymore

From there:
- either derive a new polynomial family adapted to your geometry
- create a bijective mapping from your geometry to a cylinder and modify the code to use it for the projection and the reconstruction 
- or make corrections to the current one to account for non orthogonality

You will want to work on a simple geometry to do this again. Either an ellipse or a square. Once it works there, you can move on to your complex shape

**Reply by WayneX-Wong** [reply]
Thanks for the suggestion, I may need some time to try.
